api/server/models/room.js

```diff
@@ -1,25 +1,131 @@
 var UTIL = require('../modules/generic');
-var lodash = require('lodash');
-var moment = require('moment');
-var generatePassword = require('password-generator');
+var ObjectID = require('mongodb').ObjectID;
 
-module.exports = function (User) {
+module.exports = function (Room) {
+    Room.disableRemoteMethod("create", true);
+    Room.disableRemoteMethod("upsert", true);
+    Room.disableRemoteMethod("updateAll", true);
+    Room.disableRemoteMethod("updateAttributes", false);
+    Room.disableRemoteMethod("find", true);
+    Room.disableRemoteMethod("findById", true);
+    Room.disableRemoteMethod("findOne", true);
+    Room.disableRemoteMethod("deleteById", true);
+    Room.disableRemoteMethod("login", true);
+    Room.disableRemoteMethod("logout", true);
+    Room.disableRemoteMethod("confirm", true);
+    Room.disableRemoteMethod("count", true);
+    Room.disableRemoteMethod("exists", true);
+    Room.disableRemoteMethod("resetPassword", true);
+    Room.disableRemoteMethod('createChangeStream', true);
+    Room.disableRemoteMethod('__count__accessTokens', false);
+    Room.disableRemoteMethod('__create__accessTokens', false);
+    Room.disableRemoteMethod('__delete__accessTokens', false);
+    Room.disableRemoteMethod('__destroyById__accessTokens', false);
+    Room.disableRemoteMethod('__findById__accessTokens', false);
+    Room.disableRemoteMethod('__get__accessTokens', false);
+    Room.disableRemoteMethod('__updateById__accessTokens', false);
     //********************************* START REGISTER AND LOGIN **********************************
-    User.create_private_room = function ( req, callback) {
-        
-        //console.log( req );
-        
-        //console.log( user1);
-        //console.log( user1);
-        
-        callback(null, 0, 'Invalid login', {});
-        
+    Room.create_chat_room = function ( req, room_type, room_name, chat_with, currentTimestamp, callback) {
+        if( typeof req.accessToken == 'undefined' || req.accessToken == null || req.accessToken == '' || typeof req.accessToken.userId == 'undefined' || req.accessToken.userId == '' ){
+            callback(null, 0, 'UnAuthorized', {});
+        }else if( room_type == 'private' &&  ( typeof chat_with == 'undefined' || chat_with == '') ){
+            callback(null, 0, 'Empty chat with', {});
+        }else{
+            if( typeof room_type == 'undefined' || (room_type != 'private' && room_type != 'public' ) ){
+                callback(null, 0, 'Room type must be private or public', {});
+            }else{
+                var access_token_userid = req.accessToken.userId;
+                Room.findById( access_token_userid, function(err, user) {
+                    if( err ){
+                        callback(null, 0, 'UnAuthorized', {});
+                    }else{
+                        var owner_user_id = access_token_userid;
```

Comment:
room name needs to be generated from frontend, not backend. 
room name will be made in frontend